# Illinois Men's Gymnastics Strength Test Analysis

## Introduction

A collaborative venture with coach Mike Freeze of the University of Illinois, this is an in-depth analysis of the Illini Men's Gymnastics athletes strength test performance over summer training camp in July 2023. Focusing on two tests, Drop Jumps and Isometric Mid-Thigh Pulls, we leveraged the Hawkin Dynamics system to gather multifaceted data beyond force output to better understand strength trends over time and develop actionable training recommendations to elevate athletic performance for important meets. 

In this initial analysis, the emphasis is on identifying the influential factors that predict force output, aiming to better recognize signs of strain or over-training among athletes.

## Data

Importing the dataset below.

In [11]:
# imports
import pandas as pd
import numpy as np
import glob, os

In [13]:
# read data files
folder_path_DJ = 'Drop Jump Data'
folder_path_ISO = 'Isometric Test Data' 
csv_files_DJ = glob.glob(os.path.join(folder_path_DJ, '*.csv'))
csv_files_ISO = glob.glob(os.path.join(folder_path_ISO, '*.csv'))

# generate test dataframes
DJ = pd.concat((pd.read_csv(file) for file in csv_files_DJ), ignore_index = True)
ISO = pd.concat((pd.read_csv(file) for file in csv_files_ISO), ignore_index = True)

In [14]:
# preview
DJ.head()
ISO.head()

TestId        Date      Time            Name  Segment  \
0  uoQyU9SpPmzu1Zj69KI2  07/31/2023  08:44:30    Amari Sewell      NaN   
1  B1SU3xGanPbCmJ9gO0wN  07/31/2023  08:48:28    Brandon Dang      NaN   
2  mBwLS2x05dnO3tDe9L62  07/31/2023  08:46:47   Connor McCool      NaN   
3  OC8egB9Bm2yAwanpEIGc  07/31/2023  08:53:21  Connor Micklos      NaN   
4  YgNQxOyJnHbHAkwcG0RS  07/31/2023  08:45:13   Dylan Shepard      NaN   

       Position            Type  Excluded            Tags  System Weight  ...  \
0    All-Around  Isometric Test       NaN  Mid-Thigh Pull        967.249  ...   
1            AA  Isometric Test       NaN  Mid-Thigh Pull        845.925  ...   
2  Floor, Vault  Isometric Test       NaN  Mid-Thigh Pull       1068.410  ...   
3    All-Around  Isometric Test       NaN  Mid-Thigh Pull        899.871  ...   
4            AA  Isometric Test       NaN  Mid-Thigh Pull        828.102  ...   

   Net Force at 250 ms  Relative Force at 250 ms  \
0              949.751                  198.1910   
1              595.075                  170.3461   
2             1120.590                  204.8839   
3              342.129                  138.0198   
4              528.898                  163.8687   

   Relative Force at 250 ms (BW)  RFD 0-250 ms  Left Force at 250 ms  \
0                       289.7254          3656                   948   
1                       243.5607          2288                   704   
2                       263.1965          4328                  1142   
3                       195.2677          1288                   710   
4                       231.2807          1956                   692   

   Right Force at 250 ms  Impulse 0-250ms  Net Impulse 0-250ms  \
0                    969         368.6425             127.3139   
1                    737         291.7095              80.6512   
2                   1047         373.7395             107.1712   
3                    532         324.7280             100.2102   
4                    665         311.9760             105.3646   

   Length of Pull  Time to Peak Force  
0           2.540               1.698  
1           2.389               2.368  
2           2.526               1.706  
3           2.559               1.523  
4           2.318               2.121  

[5 rows x 71 columns]

Viewing all the variables next.

In [19]:
# columns
print("Drop Jump Variables:", DJ.columns)
print("Isometric Test Variables:", ISO.columns)

Drop Jump Variables: Index(['TestId', 'Date', 'Time', 'Name', 'Segment', 'Position', 'Type',
       'Excluded', 'Tags', 'System Weight', 'Drop Height', 'Jump Height',
       'Spring Like Correlation', 'Impact Peak', 'Time to Peak Braking Force',
       'Stiffness', 'mRSI', 'RSI', 'Jump Momentum', 'Peak Braking Force',
       'Peak Relative Braking Force', 'Avg. Braking Force',
       'Avg. Relative Braking Force', 'Braking Impulse',
       'Relative Braking Impulse', 'Braking Net Impulse',
       'Relative Braking Net Impulse', 'Peak Propulsive Force',
       'Peak Relative Propulsive Force', 'Avg. Propulsive Force',
       'Avg. Relative Propulsive Force', 'Propulsive Impulse',
       'Relative Propulsive Impulse', 'Propulsive Net Impulse',
       'Relative Propulsive Net Impulse', 'Braking Phase', 'Braking Phase %',
       'Propulsive Phase', 'Propulsive Phase %', 'Time To Takeoff',
       'Flight Time', 'Contact Time', 'Positive Impulse',
       'Positive Net Impulse', 'Net Impulse 

Viewing the variable lists shows the tests measure key metrics and variations of those measures - knowing the individual factors in each test is beneficial during the modeling process. 

**Drop Jump Measures and Variables -** 

**ADD SUMMARY**

Identifiers 

* `TestId`
* `Date`
* `Time`
* `Name`
* `Type`

General measurements

* `System Weight`
* `Drop Height`
* `Jump Height`

Force

* Braking Force -  `Avg. Braking Force`, `Avg. Relative Braking Force`, `Avg. Braking RFD`, `Left Avg. Braking RFD`, `Right Avg. Braking RFD`, `L|R Peak Braking Force`, `L|R Avg. Braking Force`, `L|R Avg. Braking RFD`, `Right Force at Peak Braking Force`, `Left Force at Peak Braking Force`, `Time to Peak Braking Force`
* Propulsive Force - `Avg. Propulsive Force`, `Avg. Relative Propulsive Force`, `L|R Peak Propulsive Force`, `L|R Avg. Propulsive Force`
* Landing Force - `Avg. Landing Force`, `Relative Peak Landing Force`, `L|R Peak Landing Force`, `L|R Avg. Landing Force`, `Left Force at Peak Landing Force`, `Right Force at Peak Landing Force`

Impulse

* Braking Impulse - `Braking Impulse`, `Relative Braking Impulse`, `Braking Net Impulse`, `Relative Braking Net Impulse`, `L|R Braking Impulse Index`
* Propulsive Impulse - `Propulsive Impulse`, `Relative Propulsive Impulse`, `Propulsive Net Impulse`, `Relative Propulsive Net Impulse`, `L|R Propulsive Impulse Index`
* Landing Impulse - `L|R Landing Impulse Index`

Velocity and Power

* `Peak Velocity`
* `Takeoff Velocity`
* Braking Velocity - `Avg. Braking Velocity`
* Braking Power - `Avg. Braking Power`, `Avg. Relative Braking Power`, `Peak Braking Power`, `Peak Relative Braking Power`
* Propulsive Velocity - `Avg. Propulsive Velocity`
* Propulsive Power - `Avg. Propulsive Power`, `Avg. Relative Propulsive Power`, `Avg. Relative Propulsive Power`, `Peak Relative Propulsive Power`

Phase Timing

* `Time to Takeoff`
* `Flight Time`
* RSI - `mRSI`, `RSI`
* `Contact Time`
* Braking Phase - `Braking Phase`, `Braking Phase %`
* Propulsive Phase - `Propulsive Phase`, `Propulsive Phase %`

Other

* `Impact Peak`
* `Spring Like Correlation`
* `Countermovement Depth`
* `Jump Momentum`
* `Stiffness`

N/A

* `Segement`
* `Position`
* `Excluded`
* `Tags`

**Isometric Test Measures and Variables**

**ADD SUMMARY**

Identifiers

* `TestId`
* `Date`
* `Time`
* `Name`

General Measurements 

* `System Weight`
* `Length of Pull`

Force

* Peak Force - `Peak Force`, `Net Peak Force`, `Relative Peak Force`, `Relative Peak Force (BW)`, `L|R Peak Force`, `Left Peak Force`, `Right Peak Force`
* Force at 0 ms - `Force at 0 ms`, `Net Force at 0 ms`, `Relative Force at 0 ms`, `Relative Force at 0 ms (BW)`, `Left Force at 0 ms`, `Right Force at 0 ms`
* Force at 50 ms - `Force at 50 ms`, `Net Force at 50 ms`, `Relative Force at 50 ms`, `Relative Force at 50 ms (BW)`, `Left Force at 50 ms`, `Right Force at 50 ms`
* Force at 100 ms - `Force at 100 ms`, `Net Force at 100 ms`, `Relative Force at 100 ms`, `Relative Force at 100 ms (BW)`, `Left Force at 100 ms`, `Right Force at 100 ms`
* Force at 150 ms - `Force at 150 ms`, `Net Force at 150 ms`, `Relative Force at 150 ms`, `Relative Force at 150 ms (BW)`, `Left Force at 150 ms`, `Right Force at 150 ms`
* Force at 200 ms - `Force at 200 ms`, `Net Force at 200 ms`, `Relative Force at 200 ms`, `Relative Force at 200 ms (BW)`, `Left Force at 200 ms`, `Right Force at 200 ms`
* Force at 250 ms - `Force at 250 ms`, `Net Force at 250 ms`, `Relative Force at 250 ms`, `Relative Force at 250 ms (BW)`, `Left Force at 250 ms`, `Right Force at 250 ms`

Rate of Force Development and Impulse

* `Time to Peak Force`
* RFD 0-50 ms - `RFD 0-50 ms`, `Impulse 0-50ms`, `Net Impulse 0-50ms`
* RFD 0-100 ms - `RFD 0-100 ms`, `Impulse 0-100ms`, `Net Impulse 0-100ms`
* RFD 0-150 ms - `RFD 0-150 ms`, `Impulse 0-150ms`, `Net Impulse 0-150ms`
* RFD 0-200 ms - `RFD 0-200 ms`, `Impulse 0-200ms`, `Net Impulse 0-200ms`
* RFD 0-250 ms - `RFD 0-250 ms`, `Impulse 0-250ms`, `Net Impulse 0-250ms`

Other

* `Initiation Threshold`

N/A

* `Segment`
* `Position`
* `Excluded`
* `Tags`


## EDA